plugins/bazaar/src/components/BazaarOverviewCard/BazaarOverviewCard.tsx

```diff
@@ -0,0 +1,140 @@
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import type { BazaarProject } from '../../types';
+import { bazaarApiRef } from '../../api';
+import { Alert } from '@material-ui/lab';
+import { fetchCatalogItems } from '../../util/fetchMethods';
+import { parseBazaarProject } from '../../util/parseMethods';
+import { InfoCard } from '@backstage/core-components';
+import { bazaarPlugin } from '../../plugin';
+
+/** @public */
+export type BazaarOverviewCardProps = {
+  order: 'latest' | 'random';
+  limit: number;
+};
+
+const getUnlinkedCatalogEntities = (
+  bazaarProjects: BazaarProject[],
+  catalogEntities: Entity[],
+) => {
+  const bazaarProjectRefs = bazaarProjects.map(
+    (project: BazaarProject) => project.entityRef,
+  );
+
+  return catalogEntities.filter((entity: Entity) => {
+    return !bazaarProjectRefs?.includes(stringifyEntityRef(entity));
+  });
+};
+
+/** @public */
+export const BazaarOverviewCard = (props: BazaarOverviewCardProps) => {
+  const { order, limit } = props;
+  const bazaarApi = useApi(bazaarApiRef);
+  const catalogApi = useApi(catalogApiRef);
+  const root = useRouteRef(bazaarPlugin.routes.root);
+
+  const bazaarLink = {
+    title: 'Go to Bazaar',
+    link: root.toString(),
+  };
+
+  const [unlinkedCatalogEntities, setUnlinkedCatalogEntities] =
+    useState<Entity[]>();
+
+  const [catalogEntities, fetchCatalogEntities] = useAsyncFn(async () => {
+    return await fetchCatalogItems(catalogApi);
+  });
+
+  const [bazaarProjects, fetchBazaarProjects] = useAsyncFn(async () => {
+    const response = await bazaarApi.getProjects(limit, order);
+    return response.data.map(parseBazaarProject) as BazaarProject[];
+  });
+
+  const catalogEntityRefs = catalogEntities.value?.map((project: Entity) =>
+    stringifyEntityRef(project),
+  );
+
+  useEffect(() => {
+    const filterBrokenLinks = () => {
+      if (catalogEntityRefs) {
+        bazaarProjects.value?.forEach(async (project: BazaarProject) => {
+          if (project.entityRef) {
+            if (!catalogEntityRefs?.includes(project.entityRef)) {
+              await bazaarApi.updateProject({
+                ...project,
+                entityRef: null,
+              });
+            }
+          }
+        });
+      }
+    };
+    filterBrokenLinks();
+  }, [
+    bazaarApi,
+    bazaarProjects.value,
+    catalogEntityRefs,
+    catalogEntities.value,
+  ]);
+
+  useEffect(() => {
+    fetchCatalogEntities();
+    fetchBazaarProjects();
+  }, [fetchBazaarProjects, fetchCatalogEntities]);
+
+  useEffect(() => {
+    const unlinkedCEntities = getUnlinkedCatalogEntities(
+      bazaarProjects.value || [],
+      catalogEntities.value || [],
+    );
+
+    if (unlinkedCEntities) {
+      setUnlinkedCatalogEntities(unlinkedCEntities);
+    }
+  }, [bazaarProjects, catalogEntities]);
+
+  if (catalogEntities.error)
+    return <Alert severity="error">{catalogEntities.error.message}</Alert>;
```

Comment:
as mentioned above, if you use `<ErrorPanel error={error} />` here (and below) instead, the user will be able to see a lot of details about why things went wrong, in a way that's consistent with the rest of Backstage